modules/gdscript/gdscript_vm.cpp

```diff
@@ -686,6 +686,10 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 #else
 	OPCODE_WHILE(true) {
 #endif
+		if (!ScriptServer::is_scripting_enabled()) {
+			exit_ok = true;
```

Comment:
```suggestion
#ifdef DEBUG_ENABLED
			exit_ok = true;
#endif
```
Or move the whole section up to line 685

Reply:
Ah, I didn't catch that. I suppose this is a debug only feature, so I think I'll move it up.

Reply:
Actually, I think you do want this in non-debug builds, too.

Reply:
The break step? The `exit_ok` can't be without changing a lot of code, so then just apply my suggestion

Reply:
Yeah, I think you want to honor the scripting setting and break even in release builds even though it would be less likely to occur outside of the debugger. I just added your original suggestion.